plugin.video.embycon/resources/lib/functions.py

```diff
@@ -0,0 +1,1333 @@
+
+        info = {'title': title, 'tvshowtitle': tvshowtitle, 'mediatype': media_type}
+        log.debug('SEARCH_RESULT_ART: ' + str(art))
+        list_item.setProperty('fanart_image', art['fanart'])
+        list_item.setProperty('discart', art['discart'])
+        list_item.setArt(art)
+
+        # add count
+        list_item.setProperty('item_index', str(item_count))
+        item_count += 1
+
+        if item.get('MediaType') == 'Video':
+            total_time = str(int(float(item.get('RunTimeTicks', '0')) / (10000000 * 60)))
+            list_item.setProperty('TotalTime', str(total_time))
+            list_item.setProperty('IsPlayable', 'true')
+            list_item_url = 'plugin://plugin.video.embycon/?item_id=' + item_id + '&mode=PLAY'
+            is_folder = False
+        else:
+            item_url = ('{server}/emby/Users/{userid}' +
+                        '/items?ParentId=' + item_id +
+                        '&IsVirtualUnAired=false&IsMissing=false' +
+                        '&Fields=' + details_string +
+                        '&format=json')
+            list_item_url = 'plugin://plugin.video.embycon/?mode=GET_CONTENT&media_type={item_type}&url={item_url}'\
+                .format(item_type=item_type, item_url=urllib.quote(item_url))
+            list_item.setProperty('IsPlayable', 'false')
+            is_folder = True
+
+        menu_items = addContextMenu({}, {'id': item_id}, is_folder)
+        if len(menu_items) > 0:
+            list_item.addContextMenuItems(menu_items, True)
+
+        if (season is not None) and (episode is not None):
+            info['episode'] = episode
+            info['season'] = season
+
+        info['year'] = item.get('ProductionYear', '')
+
+        log.debug('SEARCH_RESULT_INFO: ' + str(info))
+        list_item.setInfo('Video', infoLabels=info)
+
+        item_tuple = (list_item_url, list_item, is_folder)
+        list_items.append(item_tuple)
+
+    xbmcplugin.addDirectoryItems(handle, list_items)
+    xbmcplugin.endOfDirectory(handle, cacheToDisc=False)
+
+    if progress is not None:
+        progress.update(100, i18n('done'))
+        progress.close()
+
+
+def PLAY(params, handle):
+    log.debug("== ENTER: PLAY ==")
+
+    log.debug("PLAY ACTION PARAMS: " + str(params))
+    item_id = params.get("item_id")
+
+    auto_resume = int(params.get("auto_resume", "-1"))
+    log.debug("AUTO_RESUME: " + str(auto_resume))
+
+    forceTranscode = params.get("force_transcode", None) is not None
+    log.debug("FORCE_TRANSCODE: " + str(forceTranscode))
+
+    # set the current playing item id
+    # set all the playback info, this will be picked up by the service
+    # the service will then start the playback
+
+    play_info = {}
+    play_info["item_id"] =  item_id
+    play_info["auto_resume"] = str(auto_resume)
+    play_info["force_transcode"] = forceTranscode
+    play_data = json.dumps(play_info)
+
+    home_window = HomeWindow()
```

Comment:
why do you do this instead of just setting the resolvedurl (https://codedocs.xyz/xbmc/xbmc/group__python__xbmcplugin.html#gaa03ad528583db7d5277eae3977acde0f) ?

Normally you create a listitem, set the URL in the listitem and set the resolved url using the xbmcplugin module